app/code/community/Nexcessnet/Turpentine/misc/version-4.vcl

```diff
@@ -310,6 +310,16 @@ sub vcl_hash {
             req.http.Cookie ~ "customer_group=") {
         hash_data(regsub(req.http.Cookie, "^.*?customer_group=([^;]*);*.*$", "\1"));
     }
+
+    # Some extension controllers serve different results for a single url if the
+    # request is xhr. both prototype and jquery set X-Requested-With headers
+    # on ajax requests so we can add that to the hash so that both variations
+    # are served seperately
+    if (req.http.X-Requested-With) {
+      std.log("hash_data xhr request - X-Requested-With: " + req.http.X-Requested-With);
```

Comment:
Sorry I am not 100% understanding what you are referring too.  Just the log or the whole condition?

For the log I was just kind of following the format used in the rest of the vcl. Happy to wrap it in some kind of debug condition but might need some instruction on what you need exactly.

I could also wrap the whole thing in a condition controlled by an admin setting similar to some of the other features. That way we can turn the whole thing on/off. I guess it kind of depends if the new hash will affect other installs in any way.